src/components/ApplicationMessages/index.js

```diff
@@ -18,18 +18,23 @@ const ApplicationMessageNotifier = ({ addSnackbar }) => {
         if (!hasRevalidated || isValidating) return;
         if (isLoggedIn && !data.isAdmin && !data.company?.hasFinishedRegistration && !hasShownCompleteRegistrationMessage) {
             setHasShownCompleteRegistrationMessage(true);
+
+            // Overrides key from addSnackbar()
+            const key = Date.now() + Math.random();
```

Comment:
Keys are internal to the notification system. Consumers should not know about them. I think this is a common enough scenario to make it available generaly:

Add an option to the notification, specifying if the action click should close the notification or not (I think the default should be true, but check other cases to see if it applies)

Then, somewhere [here](https://github.com/NIAEFEUP/nijobs-fe/blob/463cc3dc2738807af803c4c017b8b5da6915be06/src/components/Notifications/Notifier.js#L31), we wrap the actionHandler in a function that based on the flag calls the `closeSnackbar` (not the `handleClose`! Those are different things and the latter should only be called when the user actually closes notifications)